Tests/tests/test_TextToSpeech.cpp

```diff
@@ -0,0 +1,224 @@
+}
+
+TEST_F(TTSInitializedTest,Test_SetTTSConfiguration_Success) {
+    EXPECT_EQ(string(""), plugin->Initialize(&service));
+    
+    EXPECT_EQ(Core::ERROR_NONE, handler.Invoke(
+        connection,
+        _T("setttsconfiguration"),
+        _T("{\"language\": \"en-US\",\"voice\": \"carol\",\"ttsendpointsecured\":\"http://cdn-ec-ric-312.voice-guidance-tts.xcr.comcast.net/tts/v1/cdn/location?\",\"ttsendpointsecured\":\"http://cdn-ec-ric-312.voice-guidance-tts.xcr.comcast.net/tts/v1/cdn/location?\"}"),
+        response
+    ));
+    
+    EXPECT_EQ(response, _T("{\"TTS_Status\":0,\"success\":true}"));
+}
+
+TEST_F(TTSInitializedTest, Test_SetTTSConfiguration_Update_Success) {
+    EXPECT_EQ(string(""), plugin->Initialize(&service));
+    
+    EXPECT_EQ(Core::ERROR_NONE, handler.Invoke(
+        connection,
+        _T("setttsconfiguration"),
+        _T("{\"language\": \"en-US\",\"voice\": \"carol\",\"ttsendpoint\":\"http://cdn-ec-ric-312.voice-guidance-tts.xcr.comcast.net/tts/v1/cdn/location?\",\"ttsendpointsecured\":\"http://cdn-ec-ric-312.voice-guidance-tts.xcr.comcast.net/tts/v1/cdn/location?\",\"volume\": \"95\",\"primvolduckpercent\": \"50\",\"rate\": \"40\",\"speechrate\":\"medium\"}"),
+        response
+    ));
+
+    EXPECT_EQ(Core::ERROR_NONE, handler.Invoke(
+        connection,
+        _T("getttsconfiguration"),
+        _T(""),
+        response
+    ));
+
+    EXPECT_EQ(response, _T("{\"ttsendpoint\":\"http:\\/\\/cdn-ec-ric-312.voice-guidance-tts.xcr.comcast.net\\/tts\\/v1\\/cdn\\/location?\",\"ttsendpointsecured\":\"http:\\/\\/cdn-ec-ric-312.voice-guidance-tts.xcr.comcast.net\\/tts\\/v1\\/cdn\\/location?\",\"language\":\"en-US\",\"voice\":\"carol\",\"speechrate\":\"medium\",\"rate\":40,\"volume\":\"95\",\"TTS_Status\":0,\"success\":true}"));
+}
+
+TEST_F(TTSInitializedTest,Test_SetACL_Success) {
+    EXPECT_EQ(string(""), plugin->Initialize(&service));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Invoke(connection, _T("setACL"), _T("{\"accesslist\": [{\"method\":\"speak\",\"apps\":\"WebAPP\"}]}"), response));
+    EXPECT_EQ(response, _T("{\"success\":true}"));
+}
+
+TEST_F(TTSInitializedTest, Test_SetACL_Invalid_Callsign_Success) {
+    EXPECT_EQ(string(""), plugin->Initialize(&service));
+
+    EXPECT_EQ(Core::ERROR_NONE, handler.Invoke(connection, _T("setACL"), 
+        _T("{\"accesslist\": [{\"method\":\"speak\",\"apps\":\"WebAPP\"}]}"), response));
+
+    EXPECT_EQ(response, _T("{\"success\":true}"));
+
+    EXPECT_EQ(Core::ERROR_NONE, handler.Invoke(
+        connection,
+        _T("setttsconfiguration"), 
+        _T("{\"language\": \"en-US\",\"voice\": \"carol\",\"ttsendpointsecured\":\"http://cdn-ec-ric-312.voice-guidance-tts.xcr.comcast.net/tts/v1/cdn/location?\",\"ttsendpointsecured\":\"http://cdn-ec-ric-312.voice-guidance-tts.xcr.comcast.net/tts/v1/cdn/location?\"}"),
+        response
+    ));
+
+    EXPECT_EQ(Core::ERROR_GENERAL, handler.Invoke(connection, _T("speak"), 
```

Comment:
In this case, shouldn't we validate the return value? when speaking with 'WebAPP' the speak() API should proceed beyond access check (but ultimately fail because of the dependency in gst impl). But if a speack() request is made with some other callsign, it should fail with access check itself. Could you cover that part?